### DISCUSSION TITLE
Lambalar ve oda sorusu

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/sonsuzus#259] [Zeka Sorusu]
50 lambanın bulunduğu bir oda var bu odanın dışında bu lambaları açan 50 anahtar var. Hangi anahtarın hangi lambayı açtığı bilinmiyor. Görevimiz hangi anahtarın hangi lambayı açtığını bulmak. Anahtarların olduğu yerden lambaları göremiyoruz ancak anahtarları açtıktan sonra hangi lambaların yandığını odaya gidip görebiliyoruz. Odaya istediğimiz kadar girip çıkabiliyoruz. Odaya en az kaç kere gidip gelmemizde lambaları açan tüm anahtarları belirleyebiliriz?

**Comment by HabilCcabbar** [comment]
İlk önce iki anahtar seçeriz. Birini açarız ve belli bir süre sonra kapatırız, diğerini açık bırakırız. Sonra lamba odasına geçeriz. Bir tane açık lamba vardır. Bu lambayı belirledik. Geri kalan lambalara dokunmamız lazım. Belli bir süre açık olan diğerlerinin aksine sıcak olacaktır. Bu yöntemi uygulayarak çözebiliriz. Fakat bir sorun var. Daha önce anahtarlarına basılmış olan lambalar diğer turlarda kafamızı karıştırmasın diye o lambaları kırabiliriz. Bu sayede en az 25 turda tamamlanır.
